packages/pwa-kit-dev/src/utils/logger.js

```diff
@@ -0,0 +1,216 @@
+/*
+ * Copyright (c) 2024, Salesforce, Inc.
+ * All rights reserved.
+ * SPDX-License-Identifier: BSD-3-Clause
+ * For full license text, see the LICENSE file in the repo root or https://opensource.org/licenses/BSD-3-Clause
+ */
+
+/**
+ * Array defining the order of log levels from lowest to highest severity.
+ * The order of log levels is important. It determines the logging threshold.
+ * @type {string[]}
+ */
+export const LOG_LEVELS = ['debug', 'info', 'warn', 'error']
+
+/**
+ * Array defining the valid format for the log message.
+ * The order is not important.
+ * @type {string[]}
+ */
+const LOG_FORMATS = ['JSON', 'TEXT']
+
+// Default log level for the logging threshold. It can be any of the LOG_LEVELS values.
+const DEFAULT_LOG_LEVEL = 'info'
+
+// Default log format to use if not explicitly provided. It can be 'JSON' or 'TEXT'.
+const DEFAULT_LOG_FORMAT = 'JSON'
+
+const isServerSide = typeof window === 'undefined'
+
+// Configuration object defining logging settings for server/client environments
+const loggerConfig = isServerSide
```

Comment:
Similar to Vincents comment about controlling the configuration for client side.. what do we plan on doing here. When you have a moment, lets have conversation about that. Right now we have this global way of controlling the log level for server side, but we don't have that same mechanism for client side. And I don't think we should be using overrides in this scenario for changing the loggers log level.

Reply:
Piggybacking off this since I am also curious about how the client side works and how it's configured.

Where do the client side logs appear (I assume the browser window)? 
As for how client side logging is enabled, I assume we don't want just any shopper being able to add/enable the client side logger so depending on how we set it up, we might need a way to lock it down to authorized personnel?